components/post_view/channel_intro_message/index.ts

```diff
@@ -9,29 +9,33 @@ import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
 import {getProfilesInCurrentChannel, getCurrentUserId, getUser} from 'mattermost-redux/selectors/entities/users';
 import {get} from 'mattermost-redux/selectors/entities/preferences';
 
+import {UserProfile} from 'mattermost-redux/types/users';
+
 import {Preferences} from 'utils/constants';
 import {getDirectTeammate, getDisplayNameByUser} from 'utils/utils.jsx';
 import {getCurrentLocale} from 'selectors/i18n';
 
-import ChannelIntroMessage from './channel_intro_message.jsx';
+import {GlobalState} from 'types/store';
+
+import ChannelIntroMessage from './channel_intro_message';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     const config = getConfig(state);
     const enableUserCreation = config.EnableUserCreation === 'true';
     const isReadOnly = isCurrentChannelReadOnly(state);
     const team = getCurrentTeam(state);
     const channel = getCurrentChannel(state);
-    const teammate = getDirectTeammate(state, channel.id);
+    const teammate = getDirectTeammate(state, channel.id) as UserProfile;
     const creator = getUser(state, channel.creator_id);
 
     return {
         currentUserId: getCurrentUserId(state),
         channel,
+        fullWidth: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.CHANNEL_DISPLAY_MODE, Preferences.CHANNEL_DISPLAY_MODE_DEFAULT) === Preferences.CHANNEL_DISPLAY_MODE_FULL_SCREEN,
         locale: getCurrentLocale(state),
         channelProfiles: getProfilesInCurrentChannel(state),
         enableUserCreation,
         isReadOnly,
-        fullWidth: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.CHANNEL_DISPLAY_MODE, Preferences.CHANNEL_DISPLAY_MODE_DEFAULT) === Preferences.CHANNEL_DISPLAY_MODE_FULL_SCREEN,
```

Comment:
Just out of curiosity: is this change in the order needed? I don't have anything against it, but I guess that it's not needed because it's a named field.

Reply:
Not needed indeed. It moved it for readability purpose during work. I can revert this if you feel so.

Reply:
No need to do so, it's ok!